pkg/git/RepositoryManager.go

```diff
@@ -496,3 +523,44 @@ func transform(src *object.Commit, tag *object.Tag) (dst *Commit) {
 	}
 	return
 }
+
+
+func (impl RepositoryManagerImpl) GetLatestCommitForBranch(repository *git.Repository, branchName string) (*GitCommit, error) {
```

Comment:
duplicate code from `ChangesSinceByRepository` method. can we reuse that?

Reply:
Done.